libgnucash/engine/mocks/fake-qofquery.cpp

```diff
@@ -0,0 +1,155 @@
+#include <config.h>
+
+#include <list>
+
+#include "fake-qofquery.h"
+#include "gmock-qofbook.h"
+
+
+
+/* class QofFakeQueryPool */
+
+static class QofFakeQueryPool
+{
+public:
+    QofFakeQueryPool() {};
+
+    void addQuery(QofFakeQuery *query)
+    {
+        m_queriesNew.push_back(query);
+    }
+
+    QofFakeQuery* requestQuery(QofIdTypeConst obj_type)
+    {
+        QofFakeQuery* query = nullptr;
+
+        auto it = std::find_if(m_queriesNew.begin(), m_queriesNew.end(),
+            [obj_type](QofFakeQuery const* query) {
+                return (g_strcmp0(query->m_obj_type, obj_type) == 0);
+            });
+        if (it != m_queriesNew.end())
+        {
+            query = *it;
+            m_queriesNew.erase(it);
+            m_queriesUsed.push_back(query);
+        }
+
+        EXPECT_NE(query, nullptr);
+        return query;
+    }
+
+    bool queryUsed(QofQuery *query)
+    {
+        auto it = std::find(m_queriesUsed.begin(), m_queriesUsed.end(), (QofFakeQuery*)query);
+
+        return (it != m_queriesUsed.end());
+    }
+
+    void releaseQuery(QofFakeQuery *query)
+    {
+        auto it           = std::find(m_queriesUsed.begin(), m_queriesUsed.end(), query);
+        bool query_in_use = (it != m_queriesUsed.end());
+        ASSERT_TRUE(query_in_use);
```

Comment:
Use `ASSERT_TRUE(queryUsed(query))`.

Reply:
You're right, this has the same behaviour. My intention was to do an explicit assertion on the iterator `it` used here instead of an assertion on a function, which searches for the same iterator implicitely.

Reply:
That's just useless code duplication. It's no less explicit in the function call.

Reply:
Ok, I changed it